lib/ts/BufferWriterFormat.cc

```diff
@@ -336,70 +384,131 @@ namespace bw_fmt
       while (width-- > 0)
         w.write(spec._fill);
-      if (neg)
-        w.write(neg);
-      if (prefix1) {
-        w.write(prefix1);
-        if (prefix2)
-          w.write(prefix2);
-      }
       w.write(digits);
-      break;
-    case BWFSpec::Align::CENTER:
-      for (int i = width / 2; i > 0; --i)
-        w.write(spec._fill);
-      if (neg)
-        w.write(neg);
-      if (prefix1) {
-        w.write(prefix1);
-        if (prefix2)
-          w.write(prefix2);
-      }
-      w.write(digits);
-      for (int i = (width + 1) / 2; i > 0; --i)
-        w.write(spec._fill);
-      break;
-    case BWFSpec::Align::SIGN:
-      if (neg)
-        w.write(neg);
-      if (prefix1) {
-        w.write(prefix1);
-        if (prefix2)
-          w.write(prefix2);
+    } else { // use generic Write_Aligned
+      Write_Aligned(w,
+                    [&]() {
+                      if (prefix1) {
+                        w.write(prefix1);
+                        if (prefix2)
+                          w.write(prefix2);
+                      }
+                      w.write(digits);
+                    },
+                    spec._align, width, spec._fill, neg);
+    }
+    return w;
+  }
+
+  /// Format for floating point values. Seperates floating point into a whole number and a
+  /// fraction. The fraction is converted into an unsigned integer based on the specified
+  /// precision, spec._prec. ie. 3.1415 with precision two is seperated into two unsigned
+  /// integers 3 and 14. The different pieces are assembled and placed into the BufferWriter.
+  /// The default is two decimal places. ie. X.XX. The value is always written in base 10.
+  ///
+  /// format: whole.fraction
+  ///     or: left.right
+  BufferWriter &
+  Format_Floating(BufferWriter &w, BWFSpec const &spec, double f, bool neg_p)
+  {
+    static const ts::string_view infinity_bwf{"Inf"};
+    static const ts::string_view nan_bwf{"NaN"};
+    static const ts::string_view zero_bwf{"0"};
+    static const ts::string_view subnormal_bwf{"subnormal"};
+    static const ts::string_view unknown_bwf{"unknown float"};
+
+    // Handle floating values that are not normal
+    if (!std::isnormal(f)) {
+      ts::string_view unnormal;
+      switch (std::fpclassify(f)) {
+      case FP_INFINITE:
+        unnormal = infinity_bwf;
+        break;
+      case FP_NAN:
+        unnormal = nan_bwf;
+        break;
+      case FP_ZERO:
+        unnormal = zero_bwf;
+        break;
+      case FP_SUBNORMAL:
+        unnormal = subnormal_bwf;
+        break;
+      default:
+        unnormal = unknown_bwf;
       }
-      while (width-- > 0)
-        w.write(spec._fill);
-      w.write(digits);
-      break;
-    default:
-      if (neg)
-        w.write(neg);
-      if (prefix1) {
-        w.write(prefix1);
-        if (prefix2)
-          w.write(prefix2);
+
+      Write_Aligned(w, [&]() { w.write(unnormal); }, spec._align, spec._min - unnormal.size(), spec._fill, 0);
```

Comment:
Don't bother with `Write_Aligned` in this case. The base formatting logic will do the alignment sufficiently well, if not absolutely optimal performance. But in the case of unnormal floating point values, that's acceptable. Just `w.write(unnormal);`.

Reply:
updated to just `w.write(unnormal);`.